### DISCUSSION TITLE
Naming convention for setting names in ~/.config/scoop.json

### DISCUSSION THREAD

**Discussion by r15ch13** [ScoopInstaller/Scoop#5098] [Ideas]
Looking at https://github.com/ScoopInstaller/Scoop/blob/master/libexec/scoop-config.ps1 should make it obvious that the settings file needs a naming convention for its values. camelCase, kebab-case, snake_case, and SCREAMING_CASE are mixed together.

One idea would be to change everything to snake_case and use SCREAMING_CASE in the code so it's easier to see that a setting is used. (`get_config` and `set_config` just have to convert upper/lowercase)

```json
{
  # settings in json file
  "7zipextract_use_external": true,
  "no_junctions": false
}
```
```cmd
scoop config no_junctions true
```
```pwsh
# usage in code
$a = get_config 7ZIPEXTRACT_USE_EXTERNAL
$b = get_config NO_JUNCTIONS
```

But, now let the fight about the "correct" casing begin. :grin:

**Comment by niheaven** [comment]
So it means giving a temporary conversion function and removing it after, e.g., 3/6 months, right?

**Reply by r15ch13** [reply]
Yeah, a deprecation warning might be necessary. Don't know if someone changes settings via some automated scripts.

Also, the config functions should reject setting names that are not valid (then it could also warn about the changed names)
